### PR TITLE
S2 Menu token updates - color only

### PR DESCRIPTION
**Pull request by nabuhasan**
## Description
Added net-new tokens:

**Color:**
- menu-item-background-color-default
- menu-item-background-color-hover
- menu-item-background-color-down
- menu-item-background-color-keyboard-focus
- menu-item-background-color-disabled

These color tokens were added into both light and dark sets. And variables were created in S2/Variables. 

## Motivation and context
These changes occurred as a result of scaling things for S2 design language and experiences.

## Related issue

View [Jira task](https://jira.corp.adobe.com/browse/SDS-13388)

## Types of changes

<li> [ ] Patch (bug fixes, typos, mistakes; non-breaking change which fixes an issue) </li>
<li> [X] Minor (add a new token, changing a value, deprecating a token; non-breaking change which adds functionality) </li>
<li> [ ] Major (deleting a token, changing token value type, renaming a token by deprecating the old one; fix or feature that would cause existing functionality to change) </li>

## Checklist:

<li> [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html). </li>
<li> [X] I updated the token in all applicable sets. This applies if updating, adding, or deleting a token that has data across different sets (for example, if the value differs across color themes.) </li>

### CODE REVIEW

**Comment by mrcjhicks** [comment]
**[TESTING spectrum2-colors](https://github.com/adobe/spectrum-tokens-studio-data/actions/runs/8822751568)**

READING spectrum-tokens: `/spectrum-tokens/packages/tokens/src/`
READING   tokens studio: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-colors/`
TOKENS STUDIO DATA IS COLOR ONLY

  > WARNING - studio definition metadata missing spectrum token name: spectrum2/component,Component,menu-item,background-opacity


<details><summary>STUDIO SYNC SUMMARY</summary><p>

```
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.default
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.default
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.hover
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.hover
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.down
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.down
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.key-focus
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.key-focus
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.disabled
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.disabled
UPDATED METADATA - SPECTRUM UUID: spectrum2/component.Component.menu-item.background-opacity
```
</p></details>

<details><summary>SPECTRUM SYNC SUMMARY</summary><p>

```
NEW TOKENS: (11) menu-item-background-color-default/dark, menu-item-background-color-default/light, menu-item-background-color-hover/dark, menu-item-background-color-hover/light, menu-item-background-color-down/dark, menu-item-background-color-down/light, menu-item-background-color-keyboard-focus/dark, menu-item-background-color-keyboard-focus/light, menu-item-background-color-disabled/dark, menu-item-background-color-disabled/light, undefined
```
</p></details>

  > ERROR - studio token definition missing spectrum token name: Component/menu-item/background-opacity


TEST COMPLETE

File: src/tokens-studio/spectrum2-colors/spectrum2/component/light.json [hunk 1]
```diff
@@ -126,6 +126,64 @@
...
+            "spectrum-tokens": {
+              "uuid": "",
+              "name": "menu-item-background-color-disabled"
+            }
+          }
+        }
+      },
+      "background-opacity": {
+        "value": "0",
+        "type": "opacity"
```
Comment (karstens) [inline src/tokens-studio/spectrum2-colors/spectrum2/component/light.json:184]: @nabuhasan The opacity token should be added to the "non-color" set. We talked about this yesterday...

**Review by karstens — CHANGES_REQUESTED** [review]
This is not okay. We talked about the opacity tokens yesterday, and agreed, that they should be added to the non-color token data, where the other opacity tokens are stored, too. 
Also the opacity tokens you added here, are missing the spectrum-token entry in the json.

**Comment by mrcjhicks** [comment]
**[TESTING spectrum2-colors](https://github.com/adobe/spectrum-tokens-studio-data/actions/runs/8833338877)**

READING spectrum-tokens: `/spectrum-tokens/packages/tokens/src/`
READING   tokens studio: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-colors/`
TOKENS STUDIO DATA IS COLOR ONLY

  > WARNING - studio definition metadata missing spectrum token name: spectrum2/component,Component,menu-item,background-opacity


<details><summary>STUDIO SYNC SUMMARY</summary><p>

```
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.default
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.default
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.hover
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.hover
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.down
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.down
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.key-focus
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.key-focus
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.disabled
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.disabled
UPDATED METADATA - SPECTRUM UUID: spectrum2/component.Component.menu-item.background-opacity
```
</p></details>

<details><summary>SPECTRUM SYNC SUMMARY</summary><p>

```
NEW TOKENS: (16) neutral-subtle-background-color-default, informative-subtle-background-color-default, positive-subtle-background-color-default, notice-subtle-background-color-default, negative-subtle-background-color-default, menu-item-background-color-default/dark, menu-item-background-color-default/light, menu-item-background-color-hover/dark, menu-item-background-color-hover/light, menu-item-background-color-down/dark, menu-item-background-color-down/light, menu-item-background-color-keyboard-focus/dark, menu-item-background-color-keyboard-focus/light, menu-item-background-color-disabled/dark, menu-item-background-color-disabled/light, undefined
```
</p></details>

  > ERROR - studio token definition missing spectrum token name: Component/menu-item/background-opacity


TEST COMPLETE

**Review by lynnhao — APPROVED** [review]
Looks good to me, thank you!

File: src/tokens-studio/spectrum2-colors/spectrum2/component/dark.json [hunk 2]
```diff
@@ -126,6 +126,64 @@
...
+              "uuid": "",
+              "name": "menu-item-background-color-disabled"
+            }
+          }
+        }
+      },
+      "background-opacity": {
+        "value": "0",
+        "type": "opacity"
+      }
```
[the remark this answers is not in the document]
Comment (GarthDB) [reply src/tokens-studio/spectrum2-colors/spectrum2/component/dark.json:181]: @karstens did this get addressed? Can you add a review requesting changes if it still needs something changed?

**Review by GarthDB — CHANGES_REQUESTED** [review]

**Comment by karstens** [comment]
> > This is not okay. We talked about the opacity tokens yesterday, and agreed, that they should be added to the non-color token data, where the other opacity tokens are stored, too. Also the opacity tokens you added here, are missing the spectrum-token entry in the json.
> 
> oh whoops ~ let me go ahead and adjust that, not sure I how I missed that! @karstens

@nabuhasan no worries, thank you for fixing it so quickly.

**Review by karstens — APPROVED** [review]
Approved. @GarthDB Issue is resolved. No more changes requested.

**Review by GarthDB — APPROVED** [review]

**Comment by mrcjhicks** [comment]
**[SYNCING spectrum2-colors](https://github.com/adobe/spectrum-tokens-studio-data/actions/runs/8849416900)**

READING spectrum-tokens: `/spectrum-tokens/packages/tokens/src/`
READING   tokens studio: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-colors/`
TOKENS STUDIO DATA IS COLOR ONLY

<details><summary>STUDIO SYNC SUMMARY</summary><p>

```
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.default
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.default
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.hover
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.hover
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.down
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.down
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.key-focus
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.key-focus
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/dark.Component.menu-item.background-color.disabled
UPDATED METADATA - SPECTRUM UUID: spectrum2/component/light.Component.menu-item.background-color.disabled
```
</p></details>

<details><summary>SPECTRUM SYNC SUMMARY</summary><p>

```
NEW TOKENS: (10) menu-item-background-color-default/dark, menu-item-background-color-default/light, menu-item-background-color-hover/dark, menu-item-background-color-hover/light, menu-item-background-color-down/dark, menu-item-background-color-down/light, menu-item-background-color-keyboard-focus/dark, menu-item-background-color-keyboard-focus/light, menu-item-background-color-disabled/dark, menu-item-background-color-disabled/light
```
</p></details>

WROTE: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-colors/spectrum2/component/dark.json`
WROTE: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-colors/spectrum2/component/light.json`
WROTE: `/spectrum-tokens/packages/tokens/src/color-component.json`

SYNC COMPLETE
[Tokens Studio Pull Request](https://github.com/adobe/spectrum-tokens-studio-data/pull/120)

[Spectrum Tokens Pull Request](https://github.com/adobe/spectrum-tokens/pull/331)

**Pull request merged**